### PR TITLE
`Bishop` movement

### PR DESCRIPTION
**Pull request by swcraig**
`Bishop` includes the `DiagonalMove` module which has the list of valid diagonal vectors.  This module can be included on `Queen`, `King`, and `Bishop`. The private `movement_vectors` method on `Bishop` will be used by a future implementation in `Piece` to validate whether a desired vector can be moved along by pieces of different types.

### CODE REVIEW

File: lib/pieces/bishop.rb [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
 
 # Bishop
 class Bishop < Piece
   NOTATION = 'B'.freeze
+  include DiagonalMove
+
+  private
+
+  def movement_vectors
+    diagonal_vectors
```
Comment (jacquesporveau) [inline lib/pieces/bishop.rb]: http://blog.bigbinary.com/2012/01/08/alias-vs-alias-method.html

File: lib/diagonal_move.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      Vector[1, 1],
+      Vector[-1, 1],
+      Vector[1, -1],
+      Vector[-1, -1]
+    ].freeze
+
+  def diagonal_vectors
+    VALID_DIAGONAL_VECTORS
+  end
+end
```
Comment (natyork) [inline lib/diagonal_move.rb]: Nice! I really like this change in approach.

**Comment by jacquesporveau** [comment]
Let's get one more approval before we merge this.

File: lib/pieces/bishop.rb [hunk 3]
```diff
@@ -1,6 +1,12 @@
 require_relative 'piece'
+require_relative '../diagonal_move'
 
 # Bishop
 class Bishop < Piece
   NOTATION = 'B'.freeze
+  include DiagonalMove
+
+  alias movement_vectors diagonal_vectors
```
Comment (gregdaynes) [inline lib/pieces/bishop.rb]: Everything looks good to me, the only thing I wonder about is this `alias`

Comment (jacquesporveau) [inline lib/pieces/bishop.rb]: Yeah I recommended that. He was method wrapping. 

http://blog.bigbinary.com/2012/01/08/alias-vs-alias-method.html

**Comment by gregdaynes** [comment]
:shipit:

**Comment by swcraig** [comment]
Thank you all!

**Pull request merged**